src/Nethermind/Nethermind.Optimism/OptimismBaseFeeCalculator.cs

```diff
@@ -35,6 +37,27 @@ public UInt256 Calculate(BlockHeader parent, IEip1559Spec specFor1559)
             };
         }
 
-        return baseFeeCalculator.Calculate(parent, spec);
+        if (parent.Timestamp >= jovianTimestamp)
+        {
+            if (parent.BlobGasUsed is null)
+                throw new InvalidOperationException($"{nameof(parent.BlobGasUsed)} does not store DA footprint in post-Jovian block.");
+
+            // Override gas used for calculation if the DA footprint is larger
+            var blobGasUsed = (long)parent.BlobGasUsed;
+            if (blobGasUsed > parent.GasUsed)
+            {
+                parent = parent.Clone();
```

Comment:
Don't like additional allocation and time wasted on cloning, but simply modifying original entity seems unsafe.

Alternative can be something like adding virtual protected `Calculate` method with additional `long gasOverride` parameter to `DefaultBaseFeeCalculator`, inheriting the class instead of wrapping it, and calling the new method instead of regular `Calculate`.